cpu/stm32/stmclk/stmclk_mp1.c

```diff
@@ -0,0 +1,230 @@
+#elif CONFIG_CLOCK_APB2_DIV == 2
+#define CLOCK_APB2_DIV              (RCC_APB2DIVR_APB2DIV_1)
+#elif CONFIG_CLOCK_APB2_DIV == 4
+#define CLOCK_APB2_DIV              (RCC_APB2DIVR_APB2DIV_2)
+#elif CONFIG_CLOCK_APB2_DIV == 8
+#define CLOCK_APB2_DIV              (RCC_APB2DIVR_APB2DIV_3)
+#elif CONFIG_CLOCK_APB2_DIV == 16
+#define CLOCK_APB2_DIV              (RCC_APB2DIVR_APB2DIV_4)
+#else
+#error "Invalid APB2 prescaler value (only 1, 2, 4, 8 and 16 allowed)"
+#endif
+
+#if CONFIG_CLOCK_APB3_DIV == 1
+#define CLOCK_APB3_DIV              (RCC_APB3DIVR_APB3DIV_0)
+#elif CONFIG_CLOCK_APB3_DIV == 2
+#define CLOCK_APB3_DIV              (RCC_APB3DIVR_APB3DIV_1)
+#elif CONFIG_CLOCK_APB3_DIV == 4
+#define CLOCK_APB3_DIV              (RCC_APB3DIVR_APB3DIV_2)
+#elif CONFIG_CLOCK_APB3_DIV == 8
+#define CLOCK_APB3_DIV              (RCC_APB3DIVR_APB3DIV_3)
+#elif CONFIG_CLOCK_APB3_DIV == 16
+#define CLOCK_APB3_DIV              (RCC_APB3DIVR_APB3DIV_4)
+#else
+#error "Invalid APB3 prescaler value (only 1, 2, 4, 8 and 16 allowed)"
+#endif
+
+/* Check whether PLL must be enabled:
+  - When PLL is used as SYSCLK
+  - When PLLQ is required
+*/
+#if IS_ACTIVE(CONFIG_USE_CLOCK_PLL) || IS_ACTIVE(CONFIG_CLOCK_ENABLE_PLLQ)
+#define CLOCK_ENABLE_PLL                1
+#else
+#define CLOCK_ENABLE_PLL                0
+#endif
+
+/* Check whether HSE must be enabled:
+  - When HSE is used as SYSCLK
+  - When PLL is used as SYSCLK and the board provides HSE (since HSE will be
+    used as PLL input clock)
+*/
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSE) || \
+    (IS_ACTIVE(CONFIG_BOARD_HAS_HSE) && IS_ACTIVE(CLOCK_ENABLE_PLL))
+#define CLOCK_ENABLE_HSE                1
+#else
+#define CLOCK_ENABLE_HSE                0
+#endif
+
+/* Check whether HSI must be enabled:
+  - When HSI is used as SYSCLK
+  - When PLL is used as SYSCLK and the board doesn't provide HSE (since HSI will be
+    used as PLL input clock)
+*/
+#if IS_ACTIVE(CONFIG_USE_CLOCK_HSI) || \
+    (!IS_ACTIVE(CONFIG_BOARD_HAS_HSE) && IS_ACTIVE(CLOCK_ENABLE_PLL))
+#define CLOCK_ENABLE_HSI                1
+#else
+#define CLOCK_ENABLE_HSI                0
+#endif
+
+void stmclk_enable_hsi(void)
+{
+    RCC->OCENSETR |= RCC_OCENSETR_HSION;
+    while (!(RCC->OCRDYR & RCC_OCRDYR_HSIRDY)) {}
+}
+
+void stmclk_disable_hsi(void)
+{
+    /* we only disable the HSI clock if not used as input for the PLL and if
+     * not used directly as system clock */
+    if (CLOCK_HSE) {
+        RCC->OCENCLRR |= RCC_OCENCLRR_HSION;
+        while (!(RCC->OCRDYR & RCC_OCRDYR_HSIRDY)) {}
+    }
+}
+
+static void stmclk_enable_hse(void)
+{
+    RCC->OCENSETR |= RCC_OCENSETR_HSEON;
+    while (!(RCC->OCRDYR & RCC_OCRDYR_HSERDY)) {}
+}
+
+void stmclk_init_sysclk(void)
+{
+    /* disable any interrupts. Global interrupts could be enabled if this is
+     * called from some kind of bootloader...  */
+    unsigned is = irq_disable();
+    RCC->MC_CIFR = 0;
+
+    /* enable HSI clock for the duration of initialization */
```

Comment:
The comment doesn't really reflect what's happening after: HSI is kept enabled even after the duration the initialization.

Reply:
yes and moreover stmclk_disable_hsi() is not used either, I remove it.